packages/runner-ct/src/app/app.tsx

```diff
@@ -16,161 +17,213 @@ import { ReporterHeader } from './ReporterHeader'
 import EventManager from '../lib/event-manager'
 import { Hidden } from '../lib/Hidden'
 import { SpecList } from '../SpecList'
+import { findDOMNode } from 'react-dom'
 
 // Cypress.ConfigOptions only appears to have internal options.
 // TODO: figure out where the "source of truth" should be for
 // an internal options interface.
 export interface ExtendedConfigOptions extends Cypress.ConfigOptions {
-  projectName: string
+  projectName: string;
 }
 
 interface AppProps {
   state: State;
   // eslint-disable-next-line
-  eventManager: typeof EventManager
-  config: ExtendedConfigOptions
+  eventManager: typeof EventManager;
+  config: ExtendedConfigOptions;
+  win?: Window
 }
 
-const App: React.FC<AppProps> = observer(
-  function App (props: AppProps) {
-    const pluginRootContainer = React.useRef<null | HTMLDivElement>(null)
-
-    const { state, eventManager, config } = props
-
-    const [pluginsHeight, setPluginsHeight] = React.useState(500)
-    const [isResizing, setIsResizing] = React.useState(false)
-    const [isSpecsListOpen, setIsSpecsListOpen] = React.useState(true)
-
-    React.useEffect(() => {
-      if (pluginRootContainer.current) {
-        state.initializePlugins(config, pluginRootContainer.current)
-      }
-    }, [])
-
-    return (
-      <>
-        <main className="app-ct">
-          <div className={cs('specs-list-container', { 'specs-list-container__open': isSpecsListOpen })}>
-            <nav>
-              <a onClick={() => setIsSpecsListOpen(!isSpecsListOpen)} id="menu-toggle"
-                className="menu-toggle" aria-label="Open the menu">
-                <i className="fa fa-bars" aria-hidden="true"/>
-              </a>
-            </nav>
-            <SpecList
-              specs={state.specs}
-              selectedSpecs={state.spec ? [state.spec.absolute] : []}
-              onSelectSpec={(spec) => state.setSingleSpec(spec)}
-            />
-          </div>
-          <div className="app-wrapper">
+const DEFAULT_LEFT_SIDE_OF_SPLITPANE_WIDTH = 355
+
+const App: React.FC<AppProps> = observer(function App (props: AppProps) {
+  const pluginRootContainer = React.useRef<null | HTMLDivElement>(null)
+
+  const { state, eventManager, config, win = window } = props
+
+  const [pluginsHeight, setPluginsHeight] = React.useState(500)
+  const [leftSideOfSplitPaneWidth, setLeftSideOfSplitPaneWidth] = React.useState(DEFAULT_LEFT_SIDE_OF_SPLITPANE_WIDTH)
+  const [isResizing, setIsResizing] = React.useState(false)
+  const [isSpecsListOpen, setIsSpecsListOpen] = React.useState(true)
+  const headerRef = React.useRef()
+
+  function monitorWindowResize () {
+    const $header = $(findDOMNode(headerRef.current))
+
+    function onWindowResize () {
+      state.updateWindowDimensions({
+        windowWidth: win.innerWidth,
+        windowHeight: win.innerHeight,
+        reporterWidth: leftSideOfSplitPaneWidth,
+        headerHeight: $header.outerHeight(),
+      })
+    }
+
+    $(win)
```

Comment:
Why is this a prop? It feels weird to pass a global variable like `window` as a prop to a react component

Reply:
The idea was that it could be mocked for tests.
Since no test is written (yet) I will replace with `window`. 

Good point